core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1010,15 +1010,7 @@ class Partition(val topicPartition: TopicPartition,
       }
     }
 
-    // some delayed operations may be unblocked after HW changed
-    if (leaderHWIncremented)
-      tryCompleteDelayedRequests()
-    else {
-      // probably unblock some follower fetch requests since log end offset has been updated
-      delayedOperations.checkAndCompleteFetch()
-    }
-
-    info
+    info.copy(leaderHwChange = if (leaderHWIncremented) LeaderHwChange.Increased else LeaderHwChange.Same)
```

Comment:
With this change, `DelayedOperations.checkAndCompleteFetch()` is only used in tests. I am wondering if it can be removed. It's fine if we want to do this in a followup PR.

Unrelated to this PR, `DelayedOperations.checkAndCompleteProduce` and `DelayedOperations.checkAndCompleteDeleteRecords` seem unused. We can probably remove them in a separate PR.

Reply:
> With this change, DelayedOperations.checkAndCompleteFetch() is only used in tests. I am wondering if it can be removed. It's fine if we want to do this in a followup PR.

I will file a ticket after this PR is merged.

> Unrelated to this PR, DelayedOperations.checkAndCompleteProduce and DelayedOperations.checkAndCompleteDeleteRecords seem unused. We can probably remove them in a separate PR.

this is small change. I will remove them in this PR